client-js/app/elements/labrad-plot.ts

```diff
@@ -470,206 +787,186 @@ export class Plot extends polymer.Base {
     this.limits.yMin = isNaN(this.dataLimits.yMin) ? 0 : this.dataLimits.yMin;
     this.limits.yMax = isNaN(this.dataLimits.yMax + this.dy) ?
         0 : this.dataLimits.yMax + this.dy;
+  }
 
-    let zMin = this.dataLimits.zMin,
-        zMax = this.dataLimits.zMax;
-    if (zMin === zMax) {
-      zMin -= 1;
-      zMax += 1;
-    }
 
-    // Add a dot to the plot.
-    let w = 0,
-        h = 0;
-    const p = this;
-    switch (this.drawMode2D) {
-      case 'dots':
-        w = h = 4;
-        for (let row of data) {
-          this.chartBody
-                .append('rect')
-                .datum(row)
-                .classed('data', true)
-                .attr('x', (d) => p.xScale(d[0]))
-                .attr('y', (d) => p.yScale(d[1]) - h)
-                .attr('width', w)
-                .attr('height', h)
-                .style('fill', (d) => getColor(d[p.displaySurface], zMin, zMax));
-        }
-        break;
+  /**
+   * Projects a coordinate in graph coordinates (relative to the axes of the
+   * data) to screen coordinates (relative to the canvas). Additionally
+   * determines the width and height of the coordinate based on its type.
```

Comment:
Should indicate that this mutates the `outputObject` parameter, rather than returning a value.